### PR TITLE
feat(node/assert): Add rejects and doesNotReject

### LINKED ISSUE #872: std/node: assert.rejects is missing
`assert.rejects` is used in some [native node tests](https://github.com/denoland/deno_std/pull/695).

### PR DESCRIPTION
**Pull request by uki00a**
Closes #872.

## Changes

* Added `assert.rejects` and `assert.doesNotReject`
* Fixed `util.inspect` to enclose strings in single quotes instead of double quotes
* Introduced `validateThrownError` internal function to remove duplication between `assert.throws` and `assert.rejects`
* Introduced `gotUnwantedException` internal function to remove duplication between `assert.doesNotThrow` and `assert.doesNotReject`
* Enabled `test-assert-async.js`
* Fixed `ERR_INVALID_ARG_TYPE` to properly construct error messages

### CODE REVIEW

**Review by wperron — APPROVED** [review]
LGTM but I'd also like @kt3k's opinion before merging

File: node/assert.ts [hunk 1]
```diff
@@ -562,11 +428,370 @@ function strict(actual: unknown, message?: string | Error): asserts actual {
...
+    }
+    return Promise.reject(createAssertionError({
+      message,
+      operator: "rejects",
+      generatedMessage: true,
+    }));
+  }
+
+  // deno-lint-ignore camelcase
+  function rejects_onRejected(e: Error): void { // TODO(uki00a): In order to `test-assert-async.js` pass, intentionally adds `rejects_` as a prefix.
```
Comment (kt3k) [inline node/assert.ts:493]: 👀 👍

File: node/_errors.ts [hunk 2]
```diff
@@ -80,17 +102,88 @@ export class NodeURIError extends NodeErrorAbstraction implements URIError {
 }
 
 export class ERR_INVALID_ARG_TYPE extends NodeTypeError {
-  constructor(a1: string, a2: string | string[], a3: unknown) {
+  constructor(name: string, expected: string | string[], actual: unknown) {
```
Comment (kt3k) [inline node/_errors.ts:105]: This fix is helpful for other node compat implementations! Thanks!

**Review by kt3k — APPROVED** [review]
@uki00a LGTM. Thank you!

**Comment by uki00a** [comment]
@wperron @kt3k Thanks for the review!

**Pull request merged**